src/txmempool.cpp

```diff
@@ -1057,21 +1057,18 @@ void CTxMemPool::TrimToSize(size_t sizelimit, std::vector<COutPoint>* pvNoSpends
         CalculateDescendants(mapTx.project<0>(it), stage);
         nTxnRemoved += stage.size();
 
-        std::vector<CTransaction> txn;
         if (pvNoSpendsRemaining) {
-            txn.reserve(stage.size());
-            for (txiter iter : stage)
-                txn.push_back(iter->GetTx());
-        }
-        RemoveStaged(stage, false, MemPoolRemovalReason::SIZELIMIT);
-        if (pvNoSpendsRemaining) {
-            for (const CTransaction& tx : txn) {
-                for (const CTxIn& txin : tx.vin) {
-                    if (exists(txin.prevout.hash)) continue;
+            for (txiter iter : stage) {
```

Comment:
Not sure I understand your question... the behaviour is indeed altered since `RemoveStaged` removes stuff from mempool and thus alters the result of `exists()`.

Oh, I guess also worth mentioning: the reason `txn` had to be created in the prior version is that the contents of `stage` become invalid after `RemoveStaged`. Now that `RemoveStaged` is the final step of the loop, this is no longer a concern so `txn` doesn't need to exist now.

Reply:
Correct me if I'm wrong: an example you’re imagining is A and B are being removed from the mempool. B spends an output of A, but since we’ve already called `RemoveStaged` at this point, exists(A) will return false, and we incorrectly place that outpoint into `pvNoSpendsRemaining`.  There is no behavior change because `Uncache` will just skip the outpoint if it’s not in the coins cache.